### PR TITLE
renamed utilities/hypre_*.[c,h] to *.[c,h]

### PR DESCRIPTION
**Pull request by liruipeng**
This PR fixes name collision problem #384 introduced in v2.21 on Windows and Mac systems, where filenames are case-insensitive.

### CODE REVIEW

**Comment by rfalgout** [comment]
Hi @liruipeng .  This looks good as a quick fix for the current name collision.  Were you planning on making any other name changes?  Just curious.  Thanks!

**Review by rfalgout — APPROVED** [review]

**Comment by liruipeng** [comment]
> Hi @liruipeng . This looks good as a quick fix for the current name collision. Were you planning on making any other name changes? Just curious. Thanks!

@rfalgout Yes, I was. I just pushed the change of this file as a proof-of-concept. If it looks OK, I will make all the other changes. Thanks!

File: src/distributed_ls/pilut/DistributedMatrixPilutSolver.h [hunk 1]
```diff
@@ -9,7 +9,7 @@
 #define _DISTRIBUTED_MATRIX_PILUT_SOLVER_HEADER
 
 #include "HYPRE_config.h"
-#include "hypre_general.h"
+#include "general.h"
```
Comment (rfalgout) [inline src/distributed_ls/pilut/DistributedMatrixPilutSolver.h]: You should be able to just delete this include.  It looks like we have some a little bit of legacy header misuse.

File: src/parcsr_ls/par_amgdd_helpers.c [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 #include "_hypre_parcsr_ls.h"
 #include "_hypre_utilities.h"
-#include "hypre_hopscotch_hash.h"
+#include "hopscotch_hash.h"
```
Comment (rfalgout) [inline src/parcsr_ls/par_amgdd_helpers.c]: This needs to be added to `_hypre_utilities.h` and removed from here and elsewhere in hypre.

**Review by ulrikeyang — APPROVED** [review]
This is fine, but I am not sure how necessary this was except for the hypre_handle.c issue.
Thanks

File: src/parcsr_ls/aux_interp.c [hunk 3]
```diff
@@ -7,7 +7,7 @@
 
 #include "_hypre_parcsr_ls.h"
 #include "aux_interp.h"
-#include "hypre_hopscotch_hash.h"
+#include "hopscotch_hash.h"
```
[the remark this answers is not in the document]
Comment (liruipeng) [reply src/parcsr_ls/aux_interp.c]: @rfalgout I found `hopscotch_hash.h` is not part of `_hypre_utilities.h`. Should I put  `hopscotch_hash.h` in? Thanks!

[the remark this answers is not in the document]
Comment (liruipeng) [reply src/parcsr_ls/aux_interp.c]: @rfalgout You said yes at another place. I will add it.

**Comment by rfalgout** [comment]
Hi @ulrikeyang .  These additional changes definitely were not necessary to fix the current issue on Mac/Windows, and I would have been fine making only those changes in this PR.  However, I think the broader changes should still be made, because the approach of using the `hypre_` prefix for file naming can accidentally lead to issues like the current one.  This is almost entirely happening in the `utilities` directory, so we haven't really been doing it elsewhere (with a few exceptions).

**Comment by rfalgout** [comment]
Hi @liruipeng .  To be clear, I don't think we need to change every single instance of this naming convention in hypre.  Just finish up the changes you are making in `utilities` and let me know when you are done.  I may want to make a few additional changes and maybe even add a regression test to check for case-sensitive file naming issues to help us avoid this in the future.  Thanks!

**Comment by liruipeng** [comment]
> Hi @liruipeng . To be clear, I don't think we need to change every single instance of this naming convention in hypre. Just finish up the changes you are making in `utilities` and let me know when you are done. I may want to make a few additional changes and maybe even add a regression test to check for case-sensitive file naming issues to help us avoid this in the future. Thanks!

@rfalgout I am done. Thanks!

**Comment by rfalgout** [comment]
Hi @liruipeng .  I made a few additional changes and added a regression test to check for case-insensitive filename matches.

The following are the remaining filenames in hypre that start with `hypre_`.  For a variety of different reasons, I don't think we need to change any of them.

```
./FEI_mv/fei-hypre/hypre_cfei.cxx
./FEI_mv/fei-hypre/hypre_cfei.h
./FEI_mv/fei-hypre/hypre_lsi_amge.c
./FEI_mv/fei-hypre/hypre_lsi_ddamg.c
./FEI_mv/fei-hypre/hypre_lsi_misc.c
./FEI_mv/fei-hypre/hypre_schur_reduce.cxx
./FEI_mv/fei-hypre/hypre_slide_reduce.cxx
./blas/hypre_blas.h
./config/hypre_blas_macros.m4
./config/hypre_lapack_macros.m4
./config/hypre_macros_misc.m4
./distributed_ls/ParaSails/hypre_ParaSails.c
./distributed_ls/ParaSails/hypre_ParaSails.h
./distributed_ls/ParaSails/hypre_ParaSails.o
./docs/usr-manual/hypre_wiw.svg
./examples/docs/hypre_wiw.gif
./lapack/hypre_lapack.h
./test/hypre_ij_fortran_test.h
./test/hypre_parcsr_fortran_test.h
./test/hypre_set_precond.c
./test/hypre_sstruct_fortran_test.h
./test/hypre_struct_fortran_test.h
./test/hypre_test.h
```
 I am running the tux tests now just to be sure things are okay for both build systems.  Once that is done, I'll let you know and we can merge to master.  Thanks for taking the lead on this!

**Comment by rfalgout** [comment]
@liruipeng .  Everything looks clean on tux.  Thanks!

**Comment by liruipeng** [comment]
> @liruipeng . Everything looks clean on tux. Thanks!

Thank you @rfalgout. I am running machine-ray.sh for GPU build. Just in case. Shall we merge this if it is also clean on ray?

**Comment by rfalgout** [comment]
Yes, thanks!

**Pull request merged**